pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -301,6 +301,16 @@ func (c *Controller) reconcile() error {
 func (c *Controller) Run(stopCh <-chan struct{}) {
 	defer c.queue.ShutDown()
 
+	// If agent is running policy-only mode, it delegates routing to
+	// underlying network. Therefore it needs not know the routes to
+	// peer Pod CIDRs.
+	if c.networkConfig.TrafficEncapMode.IsNetworkPolicyOnly() {
+		klog.Infof("TrafficEncap mode is %s, skipping NodeRouteController", c.networkConfig.TrafficEncapMode)
```

Comment:
do you have to do a c.networkConfig.TrafficEncapMode.String() ?

Reply:
I think it should be OK, as "%s" formatter takes String method of variable TrafficEncapMode.